### PR TITLE
fix(components): empty icon component judgment

### LINKED ISSUE #4170: [Bug Report] Icon refactoring issue
### Element Plus version
1.2.0-beta.1

### OS/Browsers version
mac

### Vue version
3.2.20

### Reproduction Link
https://codepen.io

### Steps to reproduce
I found a lot of warnings in the documentation after the icon refactor

@adaex 
The following components will print warnings:

* [ ] select
* [ ] button
* [ ] switch
* [ ] timeline
* [x] [table](https://github.com/element-plus/element-plus/pull/4167)
* [ ] tabs
* [ ] popconfirm

### What is Expected?
No warning

### What is actually happening?
Many warnings display on the console

### PR DESCRIPTION
**Pull request by adaex**
- fix #4170

* [x] alert
* [x] input
* [x] page-header
* [x] popconfirm
* [x] result
* [x] select
* [x] select-v2
* [x] time-picker
* [x] time-select
* [x] transfer
* [x] button
* [x] switch
* [x] timeline
* [x] tabs
* [x] breadcrumb
* [x] dropdown

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/.vitepress/vitepress/components/navbar/vp-social-link.vue [hunk 1]
```diff
@@ -17,7 +17,7 @@ defineProps<{
     class="social-link"
   >
     <ElIcon :size="20">
-      <component :is="icon" />
+      <component :is="icon" v-if="icon" />
```
Comment (msidolphin) [inline docs/.vitepress/vitepress/components/navbar/vp-social-link.vue:20]: Why don't we put the v-if on ElIcon

File: docs/examples/input/mixed-input.vue [hunk 2]
```diff
@@ -30,12 +30,12 @@
...
 
 <script lang="ts">
-import { defineComponent, ref } from 'vue'
+import { defineComponent, ref, shallowRef } from 'vue'
 import { Search } from '@element-plus/icons'
 export default defineComponent({
   setup() {
     return {
-      Search,
+      Search: shallowRef(Search),
```
Comment (msidolphin) [inline docs/examples/input/mixed-input.vue:38]: There should be no need for use shallowRef

File: docs/.vitepress/vitepress/components/navbar/vp-social-link.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply docs/.vitepress/vitepress/components/navbar/vp-social-link.vue:20]: Agreed

File: docs/examples/input/mixed-input.vue (hunk 2 above)
Comment (JeremyWuuuuu) [reply docs/examples/input/mixed-input.vue:38]: Yeah, setup can return plan object, besides, we still have `raw()` method, we don't need this extra step here.

File: docs/examples/input/with-icon.vue [hunk 3]
```diff
@@ -47,8 +47,8 @@ export default defineComponent({
   },
   setup() {
     return {
-      Calendar,
-      Search,
+      Calendar: shallowRef(Calendar),
+      Search: shallowRef(Search),
```
Comment (JeremyWuuuuu) [inline docs/examples/input/with-icon.vue:51]: Same goes here not need to wrap the component with `shallowRef`

File: docs/examples/popconfirm/trigger-event.vue [hunk 4]
```diff
@@ -15,12 +15,13 @@
...
 
 <script lang="ts">
+import { shallowRef } from 'vue'
 import { InfoFilled } from '@element-plus/icons'
 
 export default {
   data() {
     return {
-      InfoFilled,
+      InfoFilled: shallowRef(InfoFilled),
```
Comment (JeremyWuuuuu) [inline docs/examples/popconfirm/trigger-event.vue:24]: Same here.

File: docs/examples/progress/custom-color.vue [hunk 5]
```diff
@@ -59,8 +59,8 @@ export default defineComponent({
       customColorMethod,
       increase,
       decrease,
-      Minus,
-      Plus,
+      Minus: shallowRef(Minus),
+      Plus: shallowRef(Plus),
```
Comment (JeremyWuuuuu) [inline docs/examples/progress/custom-color.vue:63]: Same here.

File: docs/examples/progress/dashboard-progress-bar.vue [hunk 6]
```diff
@@ -50,8 +50,8 @@ export default defineComponent({
       ...toRefs(state),
       increase,
       decrease,
-      Minus,
-      Plus,
+      Minus: shallowRef(Minus),
+      Plus: shallowRef(Plus),
```
Comment (JeremyWuuuuu) [inline docs/examples/progress/dashboard-progress-bar.vue:54]: There are lots of files with the same issue, I'm not going to point them out one by one, please address them all so we can get it merged.

File: docs/examples/progress/custom-color.vue [hunk 7]
```diff
@@ -16,7 +16,7 @@
 </template>
 
 <script lang="ts">
-import { defineComponent, reactive, toRefs } from 'vue'
+import { defineComponent, reactive, toRefs, shallowRef } from 'vue'
```
Comment (msidolphin) [inline docs/examples/progress/custom-color.vue:19]: Please remove any unused import

File: docs/examples/progress/dashboard-progress-bar.vue [hunk 8]
```diff
@@ -12,7 +12,7 @@
 </template>
 
 <script lang="ts">
-import { defineComponent, reactive, toRefs, onMounted } from 'vue'
+import { defineComponent, reactive, toRefs, onMounted, shallowRef } from 'vue'
```
Comment (msidolphin) [inline docs/examples/progress/dashboard-progress-bar.vue:15]: same here

File: docs/.vitepress/vitepress/components/navbar/vp-social-link.vue (hunk 1 above)
Comment (adaex) [reply docs/.vitepress/vitepress/components/navbar/vp-social-link.vue]: done

**Comment by adaex** [comment]
@JeremyWuuuuu  @msidolphin All done.

**Review by adaex — COMMENTED** [review]
I checked it out myself and it seems fine 😊

File: packages/components/select-v2/src/select.vue [hunk 9]
```diff
@@ -205,13 +205,13 @@
           </span>
           <span class="el-select-v2__suffix">
             <el-icon
-              v-show="!showClearBtn"
+              v-show="!showClearBtn && iconComponent"
```
Comment (msidolphin) [inline packages/components/select-v2/src/select.vue:208]: Should we use v-if here?
```vue
<el-icon
  v-if="iconComponent"
  v-show="!showClearBtn"
  :class="['el-select-v2__caret', 'el-input__icon', iconReverse]"
>
  <component :is="iconComponent" />
</el-icon>
```

Comment (adaex) [reply packages/components/select-v2/src/select.vue:208]: Oh, oh, my God. I thought he was v-if.😅

Comment (adaex) [reply packages/components/select-v2/src/select.vue]: Fixed it, as well as another place.

**Review by msidolphin — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
Let's roll them out.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**